applications/serial_lte_modem/src/main.c

```diff
@@ -124,30 +123,85 @@ static int ext_xtal_control(bool xtal_on)
 	return err;
 }
 
-/* This function is safe to call twice after reset */
-static int init_gpio(void)
+#if CONFIG_SLM_POWER_PIN >= 0
+static void power_pin_callback(const struct device *, struct gpio_callback *, uint32_t)
+{
+	LOG_INF("Power pin triggered.");
+	slm_enter_sleep(CONFIG_SLM_POWER_PIN, NRF_GPIO_PIN_SENSE_LOW);
+}
+#endif
+
+static int configure_gpio_interrupt(struct gpio_callback *gpio_cb, gpio_callback_handler_t handler,
+				    gpio_pin_t pin, gpio_flags_t flags)
+{
+	int err;
+
+	err = gpio_pin_interrupt_configure(gpio_dev, pin, flags);
+	if (err) {
+		LOG_ERR("Failed to %s for GPIO pin P0.%u. (%d)", "configure interrupt", pin, err);
+		return err;
+	}
+
+	gpio_init_callback(gpio_cb, handler, BIT(pin));
+
+	err = gpio_add_callback(gpio_dev, gpio_cb);
+	if (err) {
+		LOG_ERR("Failed to %s for GPIO pin P0.%u. (%d)", "add callback", pin, err);
+		return err;
+	}
+	LOG_DBG("Configured interrupt on GPIO pin P0.%u", pin);
+	return 0;
+}
+
+static int configure_gpio(gpio_pin_t pin, gpio_flags_t flags)
+{
+	const int err = gpio_pin_configure(gpio_dev, pin, flags);
+
+	if (err) {
+		LOG_ERR("Failed to configure GPIO pin P0.%d. (%d)", pin, err);
+		return err;
+	}
+
+	/* GPIO sense configuration has been observed to persist through pin resets. */
+	nrf_gpio_cfg_sense_set(pin, NRF_GPIO_PIN_NOSENSE);
+
+	return 0;
+}
+
+static int init_gpios(void)
 {
 	int err;
 
 	if (!device_is_ready(gpio_dev)) {
 		LOG_ERR("GPIO controller not ready");
 		return -ENODEV;
 	}
-	err = gpio_pin_configure(gpio_dev, CONFIG_SLM_WAKEUP_PIN,
-				 GPIO_INPUT | GPIO_PULL_UP | GPIO_ACTIVE_LOW);
+
+	err = configure_gpio(CONFIG_SLM_WAKEUP_PIN, GPIO_INPUT | GPIO_PULL_UP | GPIO_ACTIVE_LOW);
 	if (err) {
-		LOG_ERR("GPIO_0 config error: %d", err);
 		return err;
 	}
-#if (CONFIG_SLM_INDICATE_PIN >= 0)
-	err = gpio_pin_configure(gpio_dev, CONFIG_SLM_INDICATE_PIN,
-				 GPIO_OUTPUT_INACTIVE | GPIO_ACTIVE_LOW);
+
+#if CONFIG_SLM_INDICATE_PIN >= 0
+	err = configure_gpio(CONFIG_SLM_INDICATE_PIN, GPIO_OUTPUT_INACTIVE | GPIO_ACTIVE_LOW);
 	if (err) {
-		LOG_ERR("GPIO_0 config error: %d", err);
 		return err;
 	}
 #endif
 
+#if CONFIG_SLM_POWER_PIN >= 0
+	static struct gpio_callback gpio_cb;
+
+	err = configure_gpio(CONFIG_SLM_POWER_PIN, GPIO_INPUT | GPIO_PULL_UP | GPIO_ACTIVE_LOW);
+	if (err) {
+		return err;
+	}
+	err = configure_gpio_interrupt(&gpio_cb, power_pin_callback,
```

Comment:
I think that using edge triggered interrupt will cause our idle mode power consumption to rise for 3uA. That is the reason why the wakeup is level triggered. I can check this out.

Reply:
It would be nice if you can get power measurements to see whether it makes any difference here. And then we can easily switch to a level trigger if it helps the power consumption.

Reply:
Seems that edge triggering is still affecting this. Around 12 uA, compared to 2.7 uA with level triggering in idle mode after AT#XSLEEP=2.